### PR TITLE
Add support for the remote API

### PR DESCRIPTION
**Pull request by foreign-sub**
Add
build_key
send_key
send_macro
set_key
set_player_host

### CODE REVIEW

**Comment by Quentame** [comment]
Need this for a new [Home Assistant](https://github.com/home-assistant/core) feature @SNoof85

**Comment by SNoof85** [comment]
@Quentame do you get what's needed to test this ? I have now the Freebox Pop (server/player) then this won't work !

**Comment by Quentame** [comment]
Yep, Freebox Révolution

File: aiofreepybox/api/remote.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+                skip_auto_headers=[
+                    "Accept",
+                    "Accept-Encoding",
+                    "Content-type",
+                    "User-Agent",
+                ],
+            )
+            async with r:
+                await r.read()
+                if r.status == _PL_STATUS and r.content_length == 0:
```
Comment (Quentame) [inline aiofreepybox/api/remote.py:178]: `_PL_STATUS` const not necessary

File: aiofreepybox/api/remote.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        except (Timeout, cl_ex.ServerDisconnectedError):
+            pass
+
+        return False
+
+    def set_player_host(
+        self,
+        access_m: Optional[str] = None,
+        host: Optional[str] = None,
+        player_id: Optional[int] = None,
```
Comment (Quentame) [inline aiofreepybox/api/remote.py:189]:
```suggestion
        player_id: Optional[int] = _DEFAULT_PL_ID,
```

File: aiofreepybox/api/remote.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        host : `str`, optional
+            Default to None
+        player_id : `int`, optional
+            Default to _DEFAULT_PL_ID
+        """
+
+        self.access_mode = _DEFAULT_ACCESS_MODE if access_m is None else access_m
+        if self.access_mode == "fbxhd":
+            self.player_host = f"{_PL_HOST}"
+            f"{_DEFAULT_PL_ID if player_id is None else player_id}{_PL_DOMAIN}"
```
Comment (Quentame) [inline aiofreepybox/api/remote.py:205]:
```suggestion
            self.player_host = f"{_PL_HOST}{player_id}{_PL_DOMAIN}"
```

File: aiofreepybox/api/remote.py [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+        access_m : `str`, "local", "host", "fbxhd"
+            Default to _DEFAULT_ACCESS_MODE
+        host : `str`, optional
+            Default to None
+        player_id : `int`, optional
+            Default to _DEFAULT_PL_ID
+        """
+
+        self.access_mode = _DEFAULT_ACCESS_MODE if access_m is None else access_m
```
Comment (Quentame) [inline aiofreepybox/api/remote.py:202]:
```suggestion
        self.access_mode = _DEFAULT_ACCESS_MODE if not access_m else access_m
```

**Review by Quentame — CHANGES_REQUESTED** [review]
Some stuffs can be cleaned

File: aiofreepybox/api/remote.py [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+    key_macro_test = [{"key": "info", "long": "False"}, {"key": "info", "repeat": 0}]
+
+    def build_key(self, **key_data: Union[str, int]) -> Dict[str, Any]:
+        """
+        Build key dict
+
+        code : `str`
+        key : `str`
+        long : `str`
+        repeat : `int`
```
[the remark this answers is not in the document]
Comment (Quentame) [reply aiofreepybox/api/remote.py:85]: or Any

**Pull request closed without merging**